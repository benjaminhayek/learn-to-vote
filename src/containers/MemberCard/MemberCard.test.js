import React from 'react';
import { MemberCard } from './index.js';
import { mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow } from 'enzyme';

describe('APP', () => {
  let wrapper;
  let mockFetch;
  let mockParty;
  let congressmen;

  beforeEach(() => {
    mockFetch = jest.fn();
    mockParty = 'D';
    congressmen = {name: 'name', party: 'D', title: 'title'}
    wrapper = shallow(
      <MemberCard congressmen={congressmen}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})

describe('mapStateToProps', () => {
  it('should have access to senators, congressmen and loading', () => {
    const mockStore = {
      selected: 'undefined',
    }
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('should call dispatch when fetchCongress is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleSelected({name: 'Political Guy'});
    expect(mockDispatch).toHaveBeenCalled();
  });
})