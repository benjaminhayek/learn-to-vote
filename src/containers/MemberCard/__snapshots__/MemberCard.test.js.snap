// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberCard should match the snapshot 1`] = `
<div
  className="dem"
  onClick={[Function]}
>
  <img
    className="apple"
    src="apple.svg"
  />
  <h1
    className="name"
  >
    name
  </h1>
  <h2
    className="title"
  >
    title
  </h2>
  <h2
    className="party"
  >
    D
  </h2>
  <h2
    className="election-date"
  >
    Next Election: 
  </h2>
</div>
`;
